### PR TITLE
OM-807 | Use checkbox from HDS, make explanations flush and fix broken errors

### PR DESCRIPTION
**Pull request by Jokinen**
- Removes this project specific checkbox and uses the one from HDS instead
- Makes explanation flush for profile edit form--previously the explanation and the form were not flush on the y-axis
- Fixes broken error messages with inputs: HDS deprecated `invalidText` prop for `TextInput`

### CODE REVIEW

File: src/profile/helpers/formik.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+  return isSubmitted && isError;
+}
+
+const defaultErrorRender = (error: string) => error;
+
+export function getError<FormValues>(
+  formikProps: FormikProps<FormValues>,
+  fieldName: keyof FormValues,
+  render: (error: string) => React.ReactNode = defaultErrorRender
```
Comment (tuovinensanttu) [inline src/profile/helpers/formik.ts]: You are using `React.ReactNode` here but your import says `{ ReactNode } from 'react'`

File: src/common/explanation/Explanation.tsx [hunk 2]
```diff
@@ -7,13 +7,23 @@ type Props = {
   main: string;
   small?: string;
   className?: string;
+  variant?: 'margin' | 'flush';
```
Comment (badrange) [inline src/common/explanation/Explanation.tsx:18]: It isn't obvious to an outsider what the meaning of flush is, so some kind of comment could be helpful!

File: src/profile/components/deleteProfile/DeleteProfile.tsx [hunk 3]
```diff
@@ -87,8 +87,11 @@ function DeleteProfile(props: Props) {
 
         <Checkbox
           onChange={handleDeleteInstructions}
+          id="deleteInstructions"
           name="deleteInstructions"
-          label={
+          // eslint-disable-next-line @typescript-eslint/ban-ts-ignore
+          // @ts-ignore
```
Comment (badrange) [inline src/profile/components/deleteProfile/DeleteProfile.tsx:94]: Do we have an issue or a plan for how we can remove the need for these ts-ignores?

**Review by badrange — APPROVED** [review]

**Comment by tuovinensanttu** [comment]
Screenshot 2020-05-19 at 10 50 27

Seems like something is breaking the layout below 470px (width)

File: src/profile/components/deleteProfile/DeleteProfile.tsx (hunk 3 above)
Comment (Jokinen) [reply src/profile/components/deleteProfile/DeleteProfile.tsx:94]: I've opened an issue about this in HDS.

File: src/common/explanation/Explanation.tsx (hunk 2 above)
Comment (Jokinen) [reply src/common/explanation/Explanation.tsx:18]: Added comment.

File: src/profile/helpers/formik.ts (hunk 1 above)
Comment (Jokinen) [reply src/profile/helpers/formik.ts]: Fixed 👍

**Comment by Jokinen** [comment]
@tuovinensanttu This is how it looks for me

It looks like your browser may not support text ellipsis (or my browser, chrome, applies it automatically). But there are other inconsistencies as well--so I am not sure.

Could this relate to something along these lines: with the previous version of HDS, HDS used smaller font sizes for smaller screens and it no longer does that?

**Comment by Jokinen** [comment]
We chatted on Slack and I was able to find the root of the problem 👍

**Pull request merged**